step/step.go

```diff
@@ -0,0 +1,271 @@
+package step
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"slices"
+
+	"github.com/bitrise-io/go-steputils/v2/export"
+	"github.com/bitrise-io/go-steputils/v2/stepconf"
+	"github.com/bitrise-io/go-utils/retry"
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-io/go-xcode/v2/destination"
+)
+
+const (
+	testShardsDirKey = "BITRISE_TEST_SHARDS_PATH"
+)
+
+var ErrXcodebuildFailed = errors.New("unexpected xcodebuild failure")
+
+type Input struct {
+	ProductPath      string `env:"product_path,required"`
+	ShardCount       int    `env:"shard_count,required"`
+	ShardCalculation string `env:"shard_calculation,opt[alphabetically]"`
+	Destination      string `env:"destination,required"`
+	Verbose          bool   `env:"verbose,opt[true,false]"`
+}
+
+type Config struct {
+	ProductPath      string
+	ShardCount       int
+	ShardCalculation string
+	Destination      string
+}
+
+type Result struct {
+	TestShardsDir string
+}
+
+type Step struct {
+	inputParser    stepconf.InputParser
+	commandFactory command.Factory
+	deviceFinder   destination.DeviceFinder
+	exporter       export.Exporter
+	logger         log.Logger
+}
+
+func NewStep(
+	inputParser stepconf.InputParser,
+	commandFactory command.Factory,
+	deviceFinder destination.DeviceFinder,
+	exporter export.Exporter,
+	logger log.Logger,
+) Step {
+	return Step{
+		inputParser:    inputParser,
+		commandFactory: commandFactory,
+		deviceFinder:   deviceFinder,
+		exporter:       exporter,
+		logger:         logger,
+	}
+}
+
+func (s *Step) ProcessConfig() (*Config, error) {
+	var input Input
+	err := s.inputParser.Parse(&input)
+	if err != nil {
+		return &Config{}, err
+	}
+
+	stepconf.Print(input)
+	s.logger.EnableDebugLog(input.Verbose)
+
+	simulator, err := s.getSimulatorForDestination(input.Destination)
+	if err != nil {
+		return nil, err
+	}
+
+	s.logger.Infof("Simulator device:")
```

Comment:
Could use the somewhat newer logging introduced in xcode-test: https://github.com/bitrise-steplib/steps-xcode-test/blame/2d52e4b24129b1f4d36d85eab21266fd4da2e8db/step/step.go#L344

Reply:
Great suggestion. I updated the PR with it.